src/build/cargo.rs

```diff
@@ -521,7 +526,8 @@ pub fn make_cargo_config(build_dir: &Path, shell: Shell) -> CargoConfig {
     {
         let build_value = config_value_map.entry("build".to_owned()).or_insert(ConfigValue::Table(HashMap::new(), config_path.clone()));
 
-        let target_dir = build_dir.join("target").join("rls").to_str().unwrap().to_owned();
+        let target_dir = target_dir.map(|d| d.to_str().unwrap().to_owned())
+                .unwrap_or_else(|| build_dir.join("target").join("rls").to_str().unwrap().to_owned());
```

Comment:
Could we just use `unwrap_or` here instead?

Reply:
Why would you want to use `unwrap_or`? If you use `unwrap_or`, the default value is created every time, if you need it or not. I would propose to stay with `unwrap_or_else` here, because with this way, the default value is only created if it is required.

Reply:
Makes sense. I thought this would be also constructed lazily, but I assume we need to completely evaluate the argument to be even able to call `unwrap_or`, correct?

Reply:
Yes :) As you give the value as a normal function argument, the value is instantiated before the function is called.

Reply:
Got it, thanks!